haystack/preview/components/retrievers/memory.py

```diff
@@ -45,14 +45,29 @@ def to_dict(self) -> Dict[str, Any]:
         """
         Serialize this component to a dictionary.
         """
-        # return default_to_dict(self, ...)
+        docstore = self.document_store.to_dict()
+        docstore.pop("hash", None)
```

Comment:
We should edit `MemoryDocumentStore.to_dict()` instead of doing this.

Reply:
Ah, perfect :+1: I'm making the separate PR